reconcile/utils/gitlab_api.py

```diff
@@ -369,7 +370,7 @@ def get_merge_request_author_username(self, mr_id: int) -> str:
         return merge_request.author["username"]
 
     def get_merge_request_comments(
-        self, mr_id: int, include_description: bool = False
+        self, mr_id: int, include_description: bool = False, include_note: bool = False
```

Comment:
The result of `get_merge_request_comments` contains a list of non system notes with an optional description item, can we always have `note` key to simplify this call, so we don't need to have `include_note`?

The value can be `Optional[ProjectMergeRequestNote]`, `None` for description item, note object for real note.

Since it's optional we need to have a check before use it, can do a filter in https://github.com/app-sre/qontract-reconcile/pull/3452/files#diff-1d5ab1abb6ecb1d8c243a9e8c04d4cf6505e6be2c8932df654ea9a11c6c1b64bR409

Reply:
`include_description` is only used when need to `get_approver_decisions_from_mr_comments`, for `delete_merge_request_comments` include_description would keep as false as default, we don't need to add a filter here.

Reply:
Cool, it's OK for now, only needed when we apply typing to this file, like convert `dict` to `dataclass`, then `note` field become `Optional[ProjectMergeRequestNote]`, it requires a none check before use.